Makefile

```diff
@@ -1,11 +1,53 @@
-# Specify the name for the binaries
-NODE_DISK_MANAGER=ndm
-DOCKER_IMAGE_NAME?= openebs/node-disk-manager
-DOCKER_IMAGE_TAG?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))
+# Build the node-disk-manager image.
 
-# Use this to build only the node-disk-manager.
+build: clean vet fmt ndm version docker
 
-build: header ndm docker
+PACKAGES = $(shell go list ./... | grep -v '/vendor/')
+
+# Determine the arch/os
+ARCH:=$(shell go env GOOS)_$(shell go env GOARCH)
+
+# TAG is the tag of the docker image
+TAG?=$(shell git describe --tags --dirty --always)
```

Comment:
Can the TAG have different formats depending on:
(a) regular master build: <current-release>-githash. Example: v0.1-hash
(b) release build : <release tag>. Example: v0.1

Also, let the repo be:
openebs/ndm-ARCH (fromat).